ivy/functional/frontends/jax/nn/non_linear_activations.py

```diff
@@ -271,10 +271,13 @@ def soft_sign(x):
     return ret.astype(dtype)
 
 
-def softmax(x, /, *, axis=-1):
-    dtype = _type_conversion(x).dtype
+def softmax(x, axis=-1, where=None, initial=None):
     ret = ivy.softmax(x, axis=axis)
-    return ret.astype(dtype)
+    if initial:
```

Comment:
`initial` can exists without `where` but `where` cannot exist without `initial`, so this logic should be reversed

if `where`:
    throw error if `initial` does not exists

Reply:
The native implementation reference is here - https://jax.readthedocs.io/en/latest/_modules/jax/_src/nn/functions.html#softmax